lib/widget/entry_table.rb

```diff
@@ -23,27 +23,62 @@ class Widget::Table::EntryTable < Widget::Table
   detailed_table self
 
   def render
-    content = content_tag :table, { class: "report detail-report", id: "sortable-table" } do
-      concat head
-      concat foot
-      concat body
+    content = content_tag :div, {class: "generic-table--container"} do
+      content_tag :div, {class: "generic-table--results-container"} do
+        table = content_tag :table, {"interactive-table" => true, role:"grid", class: "generic-table", id: "sortable-table" } do
+          concat colgroup
+          concat head
+          concat foot
+          concat body
+        end
+        table + content_tag(:div, class: "generic-table--header-background"){}
+      end
     end
     # FIXME do that js-only, like a man's man
     render_widget Widget::Table::SortableInit, @subject, to: content, sort_first_row: true
     write content
   end
 
+  def colgroup
+    content_tag :colgroup do
+      Fields.map { |field| concat content_tag(:col, "highlight-col" => true) {} }
+      concat content_tag(:col, "highlight-col" => true) {}
+      concat content_tag(:col, "highlight-col" => true) {}
+      concat content_tag(:col) {}
+    end
+  end
+
   def head
     content_tag :thead do
       content_tag :tr do
-        Fields.map { |field| concat content_tag(:th) { label_for(field) } }
-        concat content_tag(:th, class: 'right') { cost_type.try(:unit_plural) || l(:units) }
-        concat content_tag(:th, class: 'right') { CostEntry.human_attribute_name(:costs) }
+        Fields.map { 
+          |field| concat content_tag(:th) {
+            content_tag(:div, class: "generic-table--sort-header-outer") {
+              content_tag(:div, class: "generic-table--sort-header") {
+                content_tag(:span, label_for(field) )
+              }
+            } 
+          } 
+        }
+        concat content_tag(:th) {
+          content_tag(:div, class: "generic-table--sort-header-outer") {
+            content_tag(:div, class: "generic-table--sort-header") {
+              content_tag(:span, cost_type.try(:unit_plural) || l(:units) )
+            }
+          }
+        }
+        concat content_tag(:th) {
+          content_tag(:div, class: "generic-table--sort-header-outer") {
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.